cl/lib/redis_utils.py

```diff
@@ -71,3 +73,67 @@ def delete_redis_semaphore(r: Union[str, Redis], key: str) -> None:
     if isinstance(r, str):
         r = get_redis_interface(r)
     r.delete(key)
+
+
+def make_update_pacer_case_id_key(court_id: str) -> str:
+    return f"update.pacer_case_id:{court_id}"
+
+
+def acquire_atomic_redis_lock(r: Redis, key, ttl: int) -> str:
+    """Acquires an atomic lock in Redis.
+
+    This method attempts to acquire a lock with the given key and TTL in Redis.
+    It uses a Lua script to ensure the lock is set atomically.
+    If the lock is already in use by another process, it retries until the
+    lock is acquired.
+
+    :param r: The Redis DB to connect to as a connection interface.
+    :param key: The key for the lock in Redis.
+    :param ttl: Time-to-live for the lock in milliseconds.
+    :return: A unique identifier for the lock.
+    """
+
+    # Lua script to acquire lock
+    lua_script = """
+    local lock_key = KEYS[1]
+    local lock_value = ARGV[1]
+    local ttl = ARGV[2]
+
+    if redis.call("SET", lock_key, lock_value, "NX", "PX", ttl) then
+        return 1
+    else
+        return 0
+    end
+    """
```

Comment:
Yeah, you're right. Acquiring the lock doesn't require a Lua script. I thought the `NX` option (used to set the key only if it doesn't exist) wasn't available in the regular Python SET method, but it is. So this method can be simplified as follows:

```
identifier = str(uuid.uuid4())
    while True:
        if r.set(key, identifier, nx=True, px=ttl):
            return identifier
        time.sleep(0.1)
```
I've updated the method to avoid using the Lua script here.

For releasing the lock, the Lua script is still required since it performs two operations: `GET` and `DEL`